### DISCUSSION TITLE
nerdctl for windows (` \\?\C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\3: The specified module could not be found`)

### DISCUSSION THREAD

**Discussion by Duttabhi** [containerd/nerdctl#2140] [Q&A]
Is nerdctl project supported as only an experimental project for windows?
I tried pulling image from local repository and it is failing stating that the specified module was not found error.

Error Message
level=fatal msg="failed to extract layer sha256..." \\\\?\\C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\3: The specified module could not be found.: unknown"

**Comment by uqix** [comment]
Same problem here, have you fixed it?

**Reply by marin2020** [reply]
Something? Anything?

**Reply by uqix** [reply]
Upgrading the Windows system fixes it.

**Reply by marin2020** [reply]
Sorry for the stupid question: Which version to upgrade to which?
In my case it is a Windows 2022 Server (latest by now)

**Reply by uqix** [reply]
We use Win Server 2019, after installing:

* windows10.0-kb5005112-x64_81d09dc6978520e1a6d44b3b15567667f83eba2c.msu
* windows10.0-kb5034127-x64_7ac59afbc502ef9dc77f7584b31110ea0b23a566.msu

it works.

BTW, it's important to disable Windows Defender (Real time protection).

**Comment by olljanat** [comment]
There might be multiple different reasons why this error appears. In my environment containers support was not enabled in Windows side. PowerShell command `Add-WindowsFeature Containers` and reboot fixed it.
